Sources/FileSystem/Configuration/TestflightConfigurationProcessor.swift

```diff
@@ -1,22 +1,83 @@
 // Copyright 2020 Itty Bitty Apps Pty Ltd
 
+import CodableCSV
 import Foundation
 import Model
+import Files
+import Yams
 
-public struct TestflightConfigurationProcessor: ResourceWriter {
+public struct TestflightConfigurationProcessor {
 
-    let path: ResourcePath
+    let appsFolderPath: String
 
-    public init(path: ResourcePath) {
-        self.path = path
+    public init(appsFolderPath: String) {
+        self.appsFolderPath = appsFolderPath
     }
 
     public func writeConfiguration(
         apps: [Model.App],
         testers: [Model.BetaTester],
         groups: [Model.BetaGroup]
-    ) {
-        // TODO: Convert models to a TestflightConfiguration and write it to disk
+    ) throws {
+        // Generate configuration
+        let groupsByApp = Dictionary(grouping: groups, by: \.app?.id)
+
+        let configurations: [TestflightConfiguration] = apps.map { app in
+            var config = TestflightConfiguration(app: app)
+
+            config.betaTesters = testers
+                .filter { tester in tester.apps.map(\.id).contains(app.id) }
+                .compactMap(FileSystem.BetaTester.init)
+
+            config.betaGroups = (groupsByApp[app.id] ?? []).map { betaGroup in
+                FileSystem.BetaGroup(
+                    betaGroup: betaGroup,
+                    betaTesters: testers.filter { $0.betaGroups.map(\.id).contains(betaGroup.id) }
+                )
+            }
+
+            return config
+        }
```

Comment:
I guess the `// Generate configuration` can be separated into a different function for easy testing purpose

Reply:
Potentially yes. Might be worth making some kind of top level file system service that the commands interact with and the processor might only need to be responsible for writing the files.